pkg/prometheus/promcfg.go

```diff
@@ -191,10 +191,28 @@ func (cg *ConfigGenerator) GenerateConfig(
 		evaluationInterval = p.Spec.EvaluationInterval
 	}
 
+	labelLimit := uint64(0)
+	if p.Spec.LabelLimit != nil {
+		labelLimit = *p.Spec.LabelLimit
+	}
+
+	labelNameLengthLimit := uint64(0)
+	if p.Spec.LabelNameLengthLimit != nil {
+		labelNameLengthLimit = *p.Spec.LabelNameLengthLimit
+	}
+
+	labelValueLengthLimit := uint64(0)
+	if p.Spec.LabelValueLengthLimit != nil {
+		labelValueLengthLimit = *p.Spec.LabelValueLengthLimit
+	}
+
 	globalItems := yaml.MapSlice{
 		{Key: "evaluation_interval", Value: evaluationInterval},
 		{Key: "scrape_interval", Value: scrapeInterval},
 		{Key: "external_labels", Value: buildExternalLabels(p)},
+		{Key: "label_limit", Value: labelLimit},
+		{Key: "label_name_length_limit", Value: labelNameLengthLimit},
+		{Key: "label_value_length_limit", Value: labelValueLengthLimit},
```

Comment:
these fields aren't available in the global configuration of Prometheus, they are only available at the scrape_config label: https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config. The enforced limits defined at the Prometheus level should be taken into account when generating the ServiceMonitors/PodMonitors/Probes configs as in https://github.com/prometheus-operator/prometheus-operator/blob/master/pkg/prometheus/promcfg.go#L658-L660. It seems that you already did that, so we can just remove the changes made in this function.